### PR TITLE
AppComplianceAutomation first GA

### PR DESCRIPTION
**Pull request by ArthurMa1978**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportPatchProperties.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        /// <summary> List of resource data. </summary>
+        public IList<ReportResourceMetadata> Resources { get; }
+        /// <summary> Report status. </summary>
+        public AppComplianceReportStatus? Status { get; }
+        /// <summary> List of report error codes. </summary>
+        public IReadOnlyList<string> Errors { get; }
+        /// <summary> Report's tenant id. </summary>
+        public Guid? TenantId { get; }
+        /// <summary> A list of comma-separated offerGuids indicates a series of offerGuids that map to the report. For example, "00000000-0000-0000-0000-000000000001,00000000-0000-0000-0000-000000000002" and "00000000-0000-0000-0000-000000000003". </summary>
+        public string OfferGuid { get; set; }
```
Comment (archerzz) [inline sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportPatchProperties.cs:111]: Looks like this should be of type `Guid`?

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportProperties.cs [hunk 2]
```diff
@@ -46,100 +46,106 @@ public partial class ReportProperties
...
+        /// <summary> List of resource data. </summary>
+        public IList<ReportResourceMetadata> Resources { get; }
+        /// <summary> Report status. </summary>
+        public AppComplianceReportStatus? Status { get; }
+        /// <summary> List of report error codes. </summary>
+        public IReadOnlyList<string> Errors { get; }
+        /// <summary> Report's tenant id. </summary>
+        public Guid? TenantId { get; }
+        /// <summary> A list of comma-separated offerGuids indicates a series of offerGuids that map to the report. For example, "00000000-0000-0000-0000-000000000001,00000000-0000-0000-0000-000000000002" and "00000000-0000-0000-0000-000000000003". </summary>
+        public string OfferGuid { get; set; }
```
Comment (archerzz) [inline sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportProperties.cs:129]: of `Guid` type?

**Review by archerzz — COMMENTED** [review]
There are many string type id properties. Please check their exact type and use `Guid` or `ResourceIdentifier` if possible. I didn't mark them all since there are too many.

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/AppComplianceReportEvidenceResource.cs [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+        /// <item>
+        /// <term>Resource</term>
+        /// <description><see cref="AppComplianceReportEvidenceResource"/></description>
+        /// </item>
+        /// </list>
+        /// </summary>
+        /// <param name="content"> Parameters for the query operation. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
+        public virtual async Task<Response<EvidenceFileDownloadResult>> DownloadAsync(EvidenceFileDownloadContent content, CancellationToken cancellationToken = default)
```
Comment (archerzz) [inline sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/AppComplianceReportEvidenceResource.cs:382]: Parameter name `EvidenceFileDownloadContent`, better to rename it to `EvidenceFileDownloadRequestContent`

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/AppComplianceReportSnapshotResource.cs [hunk 4]
```diff
@@ -180,28 +180,28 @@ public virtual Response<SnapshotResource> Get(CancellationToken cancellationToke
...
         /// </list>
         /// </summary>
         /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
         /// <param name="content"> Parameters for the query operation. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
-        public virtual async Task<ArmOperation<DownloadResponse>> DownloadAsync(WaitUntil waitUntil, SnapshotDownloadContent content, CancellationToken cancellationToken = default)
+        public virtual async Task<ArmOperation<AppComplianceDownloadResult>> DownloadAsync(WaitUntil waitUntil, SnapshotDownloadContent content, CancellationToken cancellationToken = default)
```
Comment (archerzz) [inline sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/AppComplianceReportSnapshotResource.cs:195]: Parameter name `SnapshotDownloadContent` is quite confusing, better to rename to `SnapshotDownloadRequestContent`

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceControl.cs [hunk 5]
```diff
@@ -45,50 +45,46 @@ public partial class Control
...
             ControlDescriptionHyperLink = controlDescriptionHyperLink;
             ControlStatus = controlStatus;
-            Assessments = assessments;
+            Responsibilities = responsibilities;
             _serializedAdditionalRawData = serializedAdditionalRawData;
         }
 
-        /// <summary> The Id of the control. e.g. "Operational Security#10". </summary>
+        /// <summary> The Id of the control. e.g. "Operational_Security_10". </summary>
         public string ControlId { get; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceControl.cs:76]: just string value

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceGetOverviewStatusContent.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        /// <param name="getOverviewStatusRequestType"> The resource type. </param>
+        /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
+        internal AppComplianceGetOverviewStatusContent(string getOverviewStatusRequestType, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        {
+            GetOverviewStatusRequestType = getOverviewStatusRequestType;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> The resource type. </summary>
+        public string GetOverviewStatusRequestType { get; set; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceGetOverviewStatusContent.cs:63]: No

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceControl.cs [hunk 7]
```diff
@@ -45,50 +45,46 @@ public partial class Control
...
+        /// <summary> The name of the control. e.g. "Unsupported OS and Software.". </summary>
+        public string ControlName { get; }
         /// <summary> The full name of the control. e.g. "Validate that unsupported operating systems and software components are not in use.". </summary>
         public string ControlFullName { get; }
-        /// <summary> The control type. </summary>
-        public ControlType? ControlType { get; }
         /// <summary> The control's description. </summary>
         public string ControlDescription { get; }
         /// <summary> The hyper link to the control's description'. </summary>
         public string ControlDescriptionHyperLink { get; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceControl.cs:84]: fixed

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceOnboardContent.cs [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> Initializes a new instance of <see cref="AppComplianceOnboardContent"/> for deserialization. </summary>
+        internal AppComplianceOnboardContent()
+        {
+        }
+
+        /// <summary> List of subscription ids to be onboarded. </summary>
+        public IList<string> SubscriptionIds { get; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceOnboardContent.cs:74]: SubscriptionId is string

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportCollectionGetAllOptions.cs [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        /// <summary> Number of elements to return when retrieving results. </summary>
+        public int? Top { get; set; }
+        /// <summary> OData Select statement. Limits the properties on each entry to just those requested, e.g. ?$select=reportName,id. </summary>
+        public string Select { get; set; }
+        /// <summary> The filter to apply on the operation. </summary>
+        public string Filter { get; set; }
+        /// <summary> OData order by query option. </summary>
+        public string Orderby { get; set; }
+        /// <summary> The offerGuid which mapping to the reports. </summary>
+        public string OfferGuid { get; set; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportCollectionGetAllOptions.cs:29]: Not all string, so keep it as string

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportEvidenceCollectionGetAllOptions.cs [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+        /// <summary> Number of elements to return when retrieving results. </summary>
+        public int? Top { get; set; }
+        /// <summary> OData Select statement. Limits the properties on each entry to just those requested, e.g. ?$select=reportName,id. </summary>
+        public string Select { get; set; }
+        /// <summary> The filter to apply on the operation. </summary>
+        public string Filter { get; set; }
+        /// <summary> OData order by query option. </summary>
+        public string Orderby { get; set; }
+        /// <summary> The offerGuid which mapping to the reports. </summary>
+        public string OfferGuid { get; set; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportEvidenceCollectionGetAllOptions.cs:29]: Not all string, so keep it as string

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportEvidenceProperties.cs [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+        }
+
+        /// <summary> Evidence type. </summary>
+        public AppComplianceReportEvidenceType? EvidenceType { get; set; }
+        /// <summary> The path of the file in storage. </summary>
+        public string FilePath { get; set; }
+        /// <summary> Extra data considered as evidence. </summary>
+        public string ExtraData { get; set; }
+        /// <summary> Control id. </summary>
+        public string ControlId { get; set; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportEvidenceProperties.cs:89]: string type, keep as string

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportPatchProperties.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportPatchProperties.cs:111]: Not all string, so keep it as string

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportProperties.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportProperties.cs:129]: Not all string, so keep it as string

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportCollectionGetAllOptions.cs [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+        /// <summary> The tenant id of the report creator. </summary>
+        public string ReportCreatorTenantId { get; set; }
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/Models/AppComplianceReportCollectionGetAllOptions.cs:31]: fixed

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/AppComplianceReportEvidenceResource.cs (hunk 3 above)
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/AppComplianceReportEvidenceResource.cs]: fixed

File: sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/AppComplianceReportSnapshotResource.cs (hunk 4 above)
Comment (ArthurMa1978) [reply sdk/appcomplianceautomation/Azure.ResourceManager.AppComplianceAutomation/src/Generated/AppComplianceReportSnapshotResource.cs]: fixed

**Review by archerzz — APPROVED** [review]

**Pull request merged**